deploy_stack.ps1

```diff
@@ -1,11 +1,76 @@
 #!ps1
 
+param (
+    [switch] $noAuth,
+    [switch] $n,
+    [switch] $help,
+    [switch] $h
+)
+
+if ($help -Or $h) {
+    Write-Host "Usage: "
+    Write-Host " [default]`tdeploy the OpenFaaS core services"
+    Write-Host " -noAuth [-n]`tdisable basic authentication"
+    Write-Host " -help [-h]`tdisplays this screen"
+    Exit
+}
+
 if (Get-Command docker -errorAction SilentlyContinue)
 {
-    Write-Host "Deploying stack"
+    docker node ls 2>&1 | out-null
+    if(-Not $?)
+    {
+        throw "Docker not in swarm mode, please initialise the cluster (`docker swarm init`) and retry"
+    }
+
+    Add-Type -AssemblyName System.Web
+    $secret = [System.Web.Security.Membership]::GeneratePassword(24,5)

```

Comment:
For consistency, please can you hash this value in the same way as the Linux bash script so that we don't have two sets of behaviour?

Reply:
Whats the rational for making the generated password *look* like the Linux equivalent (which is due to the method chosen for generating) - the `GeneratePassword` command is the PowerShell way of generating a valid password.

It *should* be possible to hash the password, but it complicates the script and I don't think its desirable.